docs/design.md

```diff
@@ -411,42 +415,53 @@ is choosing a timestamp guaranteed to be greater than the latest
 timestamp of any committed transaction (in absolute time). No system can
 claim consistency and fail to read already-committed data.
 
-Accomplishing this for transactions (or just single operations)
-accessing a single node is easy. The transaction supplies 0 for
-timestamp, indicating that the node should use its current time (time
-for a node is kept using a hybrid clock which combines wall time and a
-logical time). This guarantees data already committed to that node have
-earlier timestamps.
+Time for a node is maintained by a hybrid logical clock (HLC).
+The HLC time >= wall time and is potentially updated by each write at that node.
+The write's timestamp is not only used to version the data being written,
+but also potentially updates the logical time on the node. This is useful in
+guaranteeing that all data written to a node is at a timestamp < HLC time.
```

Comment:
guaranteeing that each event (e.g. a write) causes higher timestamps to be emitted for all future events on the receiving node. As an example of this, consider nodes A and B, where node A's wall time is ahead of node B's. If node A issues a write command according to its local clock at t<sub>A</sub>, node B may receive it while its local wall time t<sub>B</sub> is still smaller than the timestamp contained in the command; the command appears to happen in the future on node B. As a reaction to this, the hybrid logical clock on node B will fast-forward to a hybrid timestamp larger than t<sub>B</sub>.